naga/src/span.rs

```diff
@@ -313,12 +315,14 @@ impl<E> WithSpan<E> {
     }
 
     /// Emits a summary of the error to a string.
-    pub fn emit_to_string_with_path(&self, source: &str, path: &str) -> String
+    pub fn emit_to_string_with_path<P>(&self, source: &str, path: P) -> String
     where
         E: Error,
+        P: AsRef<std::path::Path>,
     {
         use codespan_reporting::{files, term};
 
+        let path = path.as_ref().display().to_string();
```

Comment:
suggestion: `Path::display` is essentially `Path::to_str_lossy` with extra steps. We should use that directly. Bonus: This could also let us avoid an extra allocation with `ToString::to_string`. 🙂

Reply:
We should do this for `naga/src/front/wgsl/error.rs` too then as this is just copy paste from there

Reply:
Ah, gotcha. Let's leave this as follow-up work, then.